### PR TITLE
Allow passing constants to jax_to_hlo genrule, and fix parameter ordering.

### PR DESCRIPTION
**Pull request by jlebar**
- Add new arguments `constants` and `evaled_constants` to jax_to_hlo
  genrule.  These let you curry constants into your jax function.  See
  docstrings.

- Fix ordering of HLO parameters.  JAX's "calling convention" into HLO
  for kwargs is to pass a tuple of the args sorted by their names.
  That's surprising behavior.  Instead, we now allow the user to specify
  the order of HLO parameters in the jax_to_hlo genrule.

### CODE REVIEW

**Comment by jlebar** [comment]
cc @mattjj

**Comment by jlebar** [comment]
Note to Googlers, this will require a matching internal change before merging back, cl/254507835.

**Review by mattjj — APPROVED** [review]
Awesome, LGTM! As long as you're happy with this convention of how argument order is set.

**Pull request merged**
